vips/vip-Gateway/bscmainnet.ts

```diff
@@ -0,0 +1,299 @@
+  "0x22af8a65639a351a9D5d77d5a25ea5e1Cf5e9E6b",
+  "0x08e4AFd80A5849FDBa4bBeea86ed470D697e4C54",
+];
+
+const ALL_REWARD_DISTRIBUTORS = [
+  ...REWARD_DISTRIBUTORS_DEFI,
+  ...REWARD_DISTRIBUTORS_GAMEFI,
+  ...REWARD_DISTRIBUTORS_LST,
+  ...REWARD_DISTRIBUTORS_STABLECOIN,
+  ...REWARD_DISTRIBUTORS_TRON,
+];
+
+export const vipGateway = () => {
+  const meta = {
+    version: "v2",
+    title:
+      "VIP-Gateway Update VToken and comptroller implementation in IL and Core Pool and introduces the NativeTokenGateway contract",
+    description: `
+    This VIP does the following:
+    1. Updates the implementation of all VTokens and Comptroller market facet in Core Pool
+    2. Updates the implementation of VTokens and Comptrollers in IL
+    3. Accepts the ownership of the NativeTokenGateway contract
+    4. Gives call permissions to Timelocks for seizeVenus function
+    5. Gives call permissions to Timelocks for setLastRewardingBlocks function of RewardsDistributor
+    6. Executes seizeVenus
+    7. Sets XVS and vXVS address in unitroller`,
+    forDescription: "I agree that Venus Protocol should proceed with this proposal",
+    againstDescription: "I do not think that Venus Protocol should proceed with this proposal",
+    abstainDescription: "I am indifferent to whether Venus Protocol proceeds or not",
+  };
+
+  return makeProposal(
+    [
+      {
+        target: UNITROLLER,
+        signature: "_setPendingImplementation(address)",
+        params: [DIAMOND],
+      },
+      {
+        target: DIAMOND,
+        signature: "_become(address)",
+        params: [UNITROLLER],
+      },
+      {
+        target: UNITROLLER,
+        signature: "diamondCut((address,uint8,bytes4[])[])",
+        params: [params],
+      },
+      ...CORE_MARKETS.map(vToken => {
+        return {
+          target: vToken.address,
+          signature: "_setImplementation(address,bool,bytes)",
+          params: [NEW_VBEP20_DELEGATE_IMPL, false, "0x"],
+        };
+      }),
+      {
+        target: COMPTROLLER_BEACON,
+        signature: "upgradeTo(address)",
+        params: [NEW_COMPTROLLER_IMPLEMENTATION],
+      },
+      {
+        target: VTOKEN_BEACON,
+        signature: "upgradeTo(address)",
+        params: [NEW_VTOKEN_IMPLEMENTATION],
+      },
+      {
+        target: NATIVE_TOKEN_GATEWAY,
+        signature: "acceptOwnership()",
+        params: [],
+      },
+
+      ...[NORMAL_TIMELOCK, FAST_TRACK_TIMELOCK, CRITICAL_TIMELOCK].map((timelock: string) => ({
+        target: ACM,
+        signature: "giveCallPermission(address,string,address)",
+        params: [UNITROLLER, "seizeVenus(address[],address)", timelock],
+      })),
+
+      ...ALL_REWARD_DISTRIBUTORS.map((rewardDistributor: string) => ({
+        target: ACM,
+        signature: "giveCallPermission(address,string,address)",
+        params: [rewardDistributor, "setLastRewardingBlocks(address[],uint32[],uint32[])", NORMAL_TIMELOCK],
+      })),
```

Comment:
Mapping not needed. I would use the wildcard

Reply:
Removed https://github.com/VenusProtocol/vips/pull/195/commits/45a2e5bde58b675a6ab6cf28ba03b2efd045a7e4